packages/nuxt3/src/head/runtime/lib/vueuse-head.plugin.ts

```diff
@@ -14,9 +15,26 @@ export default defineNuxtPlugin((nuxtApp) => {
     headReady = true
   })
 
-  nuxtApp._useHead = (meta: MetaObject) => {
-    const headObj = ref(meta as any)
-    head.addHeadObjs(headObj)
+  const titleTemplate = ref<MetaObject['titleTemplate']>()
+
+  nuxtApp._useHead = (meta: MetaObject | ComputedGetter<MetaObject>) => {
+    titleTemplate.value = (unref(meta) as MetaObject).titleTemplate || titleTemplate.value
+
+    const headObj = computed(() => {
+      const overrides: MetaObject = { meta: [] }
+      const val = unref(meta) as MetaObject
+      if (titleTemplate.value && 'title' in val) {
+        overrides.title = typeof titleTemplate.value === 'function' ? titleTemplate.value(val.title) : titleTemplate.value.replace(/%s/g, val.title)
+      }
+      if (val.charset) {
+        overrides.meta!.push({ key: 'charset', charset: val.charset })
+      }
+      if (val.viewport) {
+        overrides.meta!.push({ name: 'viewport', content: val.viewport })
```

Comment:
```suggestion
        overrides.meta!.push({ key: 'viewport', name: 'viewport', content: val.viewport })
```

Is it necessary too?

Reply:
`useHead` does deduplicate other meta values by name/property, so this isn't needed here.